clang/test/SemaSYCL/num_simd_work_items_device.cpp

```diff
@@ -62,13 +62,13 @@ struct TRIFuncObjBad2 {
 
 struct TRIFuncObjBad3 {
   [[intel::num_simd_work_items(3)]]     // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
-  [[cl::reqd_work_group_size(5, 3, 3)]] // expected-note{{conflicting attribute is here}}
+  [[sycl::reqd_work_group_size(5, 3, 3)]] // expected-note{{conflicting attribute is here}}
```

Comment:
[[sycl::reqd_work_group_size(5, 3, 3)]]  - this is SYCL 2020 attributes, we are not passing -sycl-std=2020 in command line. does it mean that we are seeing default to 2020 here calling clang -cc1 -fsycl-is-device?

Reply:
No, but you might be raising a good question -- should these SYCL 2020 attributes be available in earlier modes? I had assumed yes, but perhaps that's a bad assumption. Should we be warning when using the SYCL 2020 spelling but not in SYCL 2020 mode?

Reply:
>>should these SYCL 2020 attributes be available in earlier modes?

Yes, they do.

>>Should we be warning when using the SYCL 2020 spelling but not in SYCL 2020 mode?

I think we should give a warning otherwise we will have no difference between SYCL 2020 spelling [[sycl::]] vs [[intel::]]

Reply:
What's the goal of the warning? I could imagine a compatibility warning similar to `-Wpre-c++17-compat` (e.g., https://godbolt.org/z/rdbsedrGv) that users opt into when they want to know they're using a feature in a newer SYCL mode but still want to be compatible with an older SYCL mode, but that strikes me as 1) lower priority because there likely aren't many people who care about 1.2.1 compatibility, and 2) something we should be doing consistently for all attributes and other features. I could also imagine a compatibility warning similar to `-Wc++17-extensions` (e.g., https://godbolt.org/z/3jPc53f5r) that is default on to let users know they're using an extension from a newer SYCL mode while in an older SYCL mode. This one strikes me as being a potentially higher priority, but still should be done consistently for all features.

Reply:
Having thought about this a bit more, I think we should have the warnings about compatibility. I'll add those here, but we should open an issue to make sure we do this consistently for other features we've already worked on (if there are any, I'm not certain).

Reply:
>> but we should open an issue to make sure we do this consistently for other features we've already worked on

I agree

Reply:
Waiting on landing https://github.com/intel/llvm/pull/3702 before I do the additional warning work, as that sets the SYCL version to something sensible when it's not explicitly passed on cc1.